### PR TITLE
Javascript packaging support

### LINKED ISSUE #2: Automatic npm package creation
For bids-validator to be able to use this package it needs to be available on npm.

### PR DESCRIPTION
**Pull request by nellh**
Moves the Python package to a package directory to prepare it for #1 at some point. Fixes #2 by building an npm package that should work with ES modules, non-ES module browser targets, and Node.JS.

Importable as `import { rules } from 'bids-validator-common'` or `require('bids-validator-common').rules`

### CODE REVIEW

**Comment by nellh** [comment]
This doesn't automatically publish, but it makes publishing it `npm publish` or `yarn publish` and tests the build with CircleCI.

**Comment by chrisgorgo** [comment]
Overall looks good to me, but don't we need logic to publish on npm in circle.yml?

**Comment by chrisgorgo** [comment]
Sorry the original issue wasn't clear. We need an automated publication workflow like the one in bids-validator. I set up circle ci for the project with environemnt variables necessary to authenticate with npm. The rest should be more or less copy paste.

**Comment by nellh** [comment]
Okay, I'll add a publish step to CI.

**Comment by nellh** [comment]
Should be good to go now.

If you run `npm publish`, it will update the git tag and publish to npm. If you just tag the repo, it doesn't update the package.json version in git but it will publish the new tag to npm with the corrected package.json file.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
+      - run:
+          name: Sync version with git tag
+          command: npm version from-git
```
Comment (chrisgorgo) [inline .circleci/config.yml:29]: Just to double check (I could not find this online) - will this work with `v0.2.0` or does it require `0.2.0`?

Comment (nellh) [reply .circleci/config.yml:29]: It is based on the value of [this npm option](https://docs.npmjs.com/misc/config#tag-version-prefix), which defaults to 'v'. As configured here, it requires `v0.2.0` format tags.

Comment (chrisgorgo) [reply .circleci/config.yml:29]: Thanks for clarifying.

**Pull request merged**